### PR TITLE
:bug: Fix eslint setup after migration to new project layout

### PR DESCRIPTION
**Pull request by rszwajko**

### CODE REVIEW

**Review by ibolton336 — APPROVED** [review]

**Comment by rszwajko** [comment]
@ibolton336 
it might be my setup but I've encountered multiple eslint problems after the migration.
CLI included hidden vscode folders, the UI was failing to parse project (see below).
I've tweaked the config a bit and it works for me now. Please test on your env.

Screenshot from 2024-10-25 15-13-19

File: webview-ui/src/App.tsx [hunk 1]
```diff
@@ -20,7 +20,7 @@ import { vscode } from "./utils/vscode";
 import GuidedApproachWizard from "./components/GuidedApproachWizard";
 import ProgressIndicator from "./components/ProgressIndicator";
 import ViolationIncidentsList from "./components/ViolationIncidentsList";
-import { Incident, RuleSet } from "./types";
+import { Incident, RuleSet } from "@shared/types";
```
Comment (ibolton336) [inline webview-ui/src/App.tsx:23]: Oversight on my end removing the tsc compilation step from the webview build script.

**Comment by rszwajko** [comment]
@ibolton336 
If you checkout a fresh project then the problems described above are not present. Not sure what was different in my case ;/ Changes in the eslint config seem neutral.

File: eslint.config.mjs [hunk 2]
```diff
@@ -54,12 +54,7 @@ const tsConfig = {
       ecmaFeatures: {
         jsx: true,
       },
-      project: [
-        "./tsconfig.json",
-        "./shared/tsconfig.json",
-        "./vscode/tsconfig.json",
-        "./webview-ui/tsconfig.json",
-      ],
+      project: true,
```
Comment (ibolton336) [inline eslint.config.mjs:57]: Should this be necessary with latest test?

**Review by ibolton336 — CHANGES_REQUESTED** [review]

**Review by djzager — APPROVED** [review]

File: eslint.config.mjs (hunk 2 above)
Comment (rszwajko) [reply eslint.config.mjs:57]: hmm... after reverting I see the problem again (on my local repo). Only freshly cloned works for me.

Comment (rszwajko) [reply eslint.config.mjs:57]: I need to use either `project: true` or `projectService: true,`

**Review by ibolton336 — DISMISSED** [review]

File: eslint.config.mjs (hunk 2 above)
Comment (ibolton336) [reply eslint.config.mjs:57]: Ah I was able to reproduce this after installing the proper eslint extensions.

**Review by ibolton336 — APPROVED** [review]
LGTM after testing the setting on a fedora machine with eslint / eslint-prettier extensions.

**Pull request merged**